apps/readest-app/src/components/PaperNoteModal.tsx

```diff
@@ -0,0 +1,332 @@
+      return;
+    }
+
+    setIsTesting(true);
+    setError('');
+    setSuccess('');
+
+    try {
+      // 这里使用fetch测试连接
+      await fetch(`http://${host}:${port}/ping`, {
+        method: 'HEAD',
+        mode: 'no-cors', // 避免CORS问题
+        cache: 'no-store',
+      });
+
+      // 只要能收到响应就认为连接成功
+      setSuccess(_('Connection successful'));
+    } catch (err) {
+      setError(_('Failed to connect to server. Please check address and port.'));
+    } finally {
+      setIsTesting(false);
+    }
+  };
+
+  const handleSubmit = async () => {
+    // if (!success && !confirm(_('Server connection not verified. Continue anyway?'))) {
+    //   return;
+    // }
+    if (!sideBarBookKey) {
+      eventDispatcher.dispatch('toast', {
+        type: 'error',
+        message: _('No book selected'),
+        className: 'whitespace-nowrap',
+        timeout: 2000,
+      });
+      return;
+    }
+    const bookData = getBookData(sideBarBookKey)!;
+    const { bookDoc, book } = bookData;
+    if (!bookDoc || !book || !bookDoc.toc) return;
+    const config = getConfig(sideBarBookKey)!;
+    const { booknotes: allNotes = [] } = config;
+    const booknotes = allNotes.filter((note) => !note.deletedAt);
+    if (booknotes.length === 0) {
+      eventDispatcher.dispatch('toast', {
+        type: 'info',
+        message: _('No annotations to export'),
+        className: 'whitespace-nowrap',
+        timeout: 2000,
+      });
+      return;
+    }
+
+    const booknoteGroups: { [href: string]: BooknoteGroup } = {};
+    for (const booknote of booknotes) {
+      const tocItem = findTocItemBS(bookDoc.toc ?? [], booknote.cfi);
+      const href = tocItem?.href || '';
+      const label = tocItem?.label || '';
+      const id = tocItem?.id || 0;
+
+      if (!booknoteGroups[href]) {
+        booknoteGroups[href] = { id, href, label, booknotes: [] };
+      }
+      booknoteGroups[href].booknotes.push(booknote);
+    }
+
+    Object.values(booknoteGroups).forEach((group) => {
+      group.booknotes.sort((a, b) => CFI.compare(a.cfi, b.cfi));
+    });
+
+    const sortedGroups = Object.values(booknoteGroups).sort((a, b) => a.id - b.id);
+
+    setIsLoading(true);
+    setError('');
+
+    try {
+      const paperNotes: PaperNoteApi = {
+        title: book.title,
+        type: bookInfo.type,
+        author: book.author,
+        cover: book.coverImageUrl?.substring(5),
+        translator: '',
+        publisher: '',
+        publishDate: 0, // 默认为0
+        isbn: '',
+        locationUnit: bookInfo.locationUnit,
+        entries: sortedGroups.flatMap((group) =>
+          group.booknotes.map((note) => ({
+            page: 0, // 默认为0 暂时不使用
+            text: note.text,
+            note: note.note,
+            chapter: group.label || _('Untitled'),
+            time: Math.floor(note.createdAt / 1000),
+          })),
+        ),
+      };
+
+      const response = await fetch(`http://${host}:${port}/send`, {
```

Comment:
Connecting to arbitrary url is not allowed by default in Tauri Apps according to current security model.